funcr/funcr.go

```diff
@@ -269,29 +270,27 @@ func (l fnlogger) caller() callerID {
 	// +1 for this frame, +1 for logr itself.
 	// FIXME: Maybe logr should offer a clue as to how many frames are
 	// needed here?  Or is it part of the contract to LogSinks?
-	_, file, line, ok := runtime.Caller(framesToCaller() + 2)
+	_, file, line, ok := runtime.Caller(framesToCaller() + l.depth + 2)
```

Comment:
Since the `framesToCaller()` thing is a common reoccurrence, perhaps provide a public helper function for this, and to get the extra number of stack frames added by the concrete `Logger`. This prevents log sinks from having to depend on implementation details of the `Logger` to accurately determine the caller frame.

Reply:
resolving this, discussing below